### DISCUSSION TITLE
Nodemailer not working, edit not working

### DISCUSSION THREAD

**Discussion by Comandante1977** [outline/outline#3676] [Self Hosting] [answered]
After hours of magic my slack admin can login, but he cant edit anything. Only titles. Cant upload anything. 
And invitation are not send because I've got :
Error: Connection timeout
    at SMTPConnection._formatError (/Distr/node_modules/nodemailer/lib/smtp-connection/index.js:784:19)
    at SMTPConnection._onError (/Distr/node_modules/nodemailer/lib/smtp-connection/index.js:770:20)
    at Timeout.<anonymous> (/Distr/node_modules/nodemailer/lib/smtp-connection/index.js:229:22)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7) {
  code: 'ETIMEDOUT',
  command: 'CONN'
Tried with gmail, mail.ru and even with my own mail server. No results - connection timeout

Here my .env file:

SECRET_KEY=secret here
UTILS_SECRET=secret here

DATABASE_URL=postgres://wikiadm:pass@localhost:5432/outline
DATABASE_URL_TEST=postgres://postgres:pass@localhost:5432/outline-test
DATABASE_CONNECTION_POOL_MIN=
DATABASE_CONNECTION_POOL_MAX=

PGSSLMODE=disable

REDIS_URL=redis://localhost:6379

URL=https://wiki.fourup.io
PORT=2999

COLLABORATION_URL=

AWS_ACCESS_KEY_ID=login
AWS_SECRET_ACCESS_KEY=pass
AWS_REGION=eu-central-1
AWS_S3_ACCELERATE_URL=
AWS_S3_UPLOAD_BUCKET_URL=https://wikidata.fourup.io
AWS_S3_UPLOAD_BUCKET_NAME=wikibuck
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private

SLACK_CLIENT_ID=ID here
SLACK_CLIENT_SECRET=Secret here

GOOGLE_CLIENT_ID=
GOOGLE_CLIENT_SECRET=

AZURE_CLIENT_SECRET=
AZURE_RESOURCE_APP_ID=

OIDC_CLIENT_ID=
OIDC_CLIENT_SECRET=
OIDC_AUTH_URI=
OIDC_TOKEN_URI=
OIDC_USERINFO_URI=
OIDC_USERNAME_CLAIM=preferred_username
OIDC_DISPLAY_NAME=OpenID
OIDC_SCOPES=openid profile email

SSL_KEY=
SSL_CERT=

CDN_URL=

FORCE_HTTPS=false

ENABLE_UPDATES=true

WEB_CONCURRENCY=1

MAXIMUM_IMPORT_SIZE=5120000

DEBUG=http,emails,mailer

SLACK_VERIFICATION_TOKEN=Token here
SLACK_APP_ID=ID here
SLACK_MESSAGE_ACTIONS=true

GOOGLE_ANALYTICS_ID=
SENTRY_DSN=

SMTP_HOST=Host here
SMTP_PORT=465
SMTP_USERNAME=am@myvd.ml
SMTP_PASSWORD=pass here
SMTP_FROM_EMAIL=am@myvd.ml
SMTP_REPLY_EMAIL=am@myvd.ml
SMTP_TLS_CIPHERS=
SMTP_SECURE=true

DEFAULT_LANGUAGE=en_US

Any ideas where I\m wrong?

And once again - I'm not using docker

**Comment by almereyda** [comment, marked as answer]
Let me ask these questions first, before trying to draw any conclusions:

- When you used your mail server, could you see a connection attempt in the logs?

- May it be that access to port 465 is blocked in general at the network level?

- Would a STARTTLS connection at port 587 work as expected (`SMTP_SECURE=false`)?

- Do the hosts have working name resolution for the hosts acting as `SMTP_HOST`

Additionally, `SMTPConnection._formatError` might hint at some syntactic problem in the `SMTP_HOST` environment variable, instead.

**Reply by Comandante1977** [reply]
You are on right way! :) Just investigated that Hetzner (my VDS there) block outgoing ports 25 and 465. Now trying to play with 587.

**Reply by almereyda** [reply]
Cool! Further details about SMTP had been covered in

- #3284

**Reply by Comandante1977** [reply]
Well ok. I've configured to use mail.ru service and can enter Outline with magic link. Ok. 
But inside I can only create doc and edit title, but no body text. Strange....
It seems that internal Outline editor is not working for me...
![image](https://user-images.githubusercontent.com/45628795/174669004-e16c1f59-f89c-4c3d-bd6d-96e13e95960e.png)
Cant write anything

**Reply by almereyda** [reply]
Glad to hear it worked out!

Feel free to mark this discussion as answered, to make it easier for others to find our considerations.

Then to follow up on the other question, would you mind opening a separate discussion for that, in so we can resolve the situations independently?
